components/ILIAS/Mail/classes/class.ilMailFormUploadHandlerGUI.php

```diff
@@ -0,0 +1,124 @@
+use ILIAS\FileUpload\Handler\BasicFileInfoResult;
+use ILIAS\FileUpload\Handler\BasicHandlerResult;
+use ILIAS\FileUpload\Handler\FileInfoResult;
+use ILIAS\FileUpload\Handler\HandlerResult as HandlerResultInterface;
+use ILIAS\ResourceStorage\Identification\ResourceIdentification;
+use ILIAS\ResourceStorage\Consumer\InlineConsumer;
+use ILIAS\ResourceStorage\Consumer\FileStreamConsumer;
+
+class ilMailFormUploadHandlerGUI extends AbstractCtrlAwareUploadHandler
+{
+    private readonly ILIAS\ResourceStorage\Services $storage;
+    private readonly ilMailAttachmentStakeholder $stakeholder;
+    private readonly ilLanguage $lng;
+
+    public function __construct()
+    {
+        global $DIC;
+        parent::__construct();
+        $this->storage = $DIC->resourceStorage();
+        $this->stakeholder = new ilMailAttachmentStakeholder();
+        $this->lng = $DIC->language();
+    }
+
+    protected function getUploadResult(): HandlerResultInterface
+    {
+        $this->upload->process();
+        /**
+         * @var UploadResult $result
+         */
+        $array = $this->upload->getResults();
+        $result = end($array);
+        if ($result instanceof UploadResult && $result->isOK()) {
+            $i = $this->storage->manage()->upload($result, $this->stakeholder);
+            $status = HandlerResultInterface::STATUS_OK;
+            $identifier = $i->serialize();
+            $message = $this->lng->txt('upload_ok');
+        } else {
+            $status = HandlerResultInterface::STATUS_FAILED;
+            $identifier = '';
+            $message = $result->getStatus()->getMessage();
+        }
+
+        return new BasicHandlerResult($this->getFileIdentifierParameterName(), $status, $identifier, $message);
+    }
+
+    protected function getRemoveResult(string $identifier): HandlerResultInterface
+    {
+        $id = $this->storage->manage()->find($identifier);
+        if ($id !== null) {
+            $this->storage->manage()->remove($id, $this->stakeholder);
+            return new BasicHandlerResult($this->getFileIdentifierParameterName(), HandlerResultInterface::STATUS_OK, $identifier, 'file deleted');
+        } else {
+            return new BasicHandlerResult($this->getFileIdentifierParameterName(), HandlerResultInterface::STATUS_FAILED, $identifier, 'file not found');
+        }
+    }
+
+    public function getInfoResult(string $identifier): FileInfoResult
+    {
+        $id = $this->storage->manage()->find($identifier);
+        if ($id === null) {
+            return new BasicFileInfoResult($this->getFileIdentifierParameterName(), 'unknown', 'unknown', 0, 'unknown');
+        }
+        $r = $this->storage->manage()->getCurrentRevision($id)->getInformation();
+
+        return new BasicFileInfoResult($this->getFileIdentifierParameterName(), $identifier, $r->getTitle(), $r->getSize(), $r->getMimeType());
+    }
+
+    /**
+     * @param array<string, mixed> $file_ids
+     * @return array<BasicFileInfoResult>
+     */
```

Comment:
Sorry, I just saw that the type is already specified in the base class `\ILIAS\UI\Component\Input\Field\UploadHandler::getInfoForExistingFiles`.

My bad ;-).

```suggestion
```